inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -188,60 +207,72 @@ void MKLDNNPadNode::createPrimitive() {
         params.dstStrides[i] = params.dstStrides[i + 1] * params.dstDims[i + 1];
     }
 
-    if (getParentEdgeAt(0)->getMemory().getDesc().hasLayoutType(LayoutType::nCsp16c) ||
-            getParentEdgeAt(0)->getMemory().getDesc().hasLayoutType(LayoutType::nCsp8c)) {
-        padsBegin[1] /= params.srcDims[params.srcDims.size() - 1];
-        padsEnd[1] /= params.srcDims[params.srcDims.size() - 1];
-        padsBegin.push_back(0);
-        padsEnd.push_back(0);
-    } else {
-        auto order = inBlkDesc->getOrder();
-        std::vector<unsigned int> newPadsBegin(padsBegin.size(), 0), newPadsEnd(padsEnd.size(), 0);
-        for (size_t i = 0; i < padsBegin.size(); ++i) {
-            newPadsBegin[i] = padsBegin[order[i]];
-            newPadsEnd[i] = padsEnd[order[i]];
+    // pads are constant, so we can calculate new collapsing pads for first target dimensions and use it for the next dimensions
```

Comment:
please see comments: https://github.com/openvinotoolkit/openvino/pull/8375#discussion_r742024403

Reply:
I decided to leave calculations of new pads and indexes because if pads are constant (now we support inly this case), we can calculate collapsing pads once for all shapes. Other the calculations for dims, strides are in executor. Thank you